lms/static/js/utils/dropdown.js

```diff
@@ -0,0 +1,203 @@
+
+                    } else if (focused.is('.has-dropdown')) {
+                        // Key handlers for when the button that opens the menu has focus
+                        _dropdown.handlerIsButton(keyCode, focused);
+
+                    } else if (focused.is('.dropdown-menu')) {
+                        // Key handlers for when the menu itself has focus, before an item within it receives focus
+                        menu = focused.closest('.dropdown-menu');
+                        _dropdown.handlerIsMenu(keyCode, menu);
+                    }
+                });
+            },
+
+            previousMenuItemLink: function(focused, menu) {
+                var items = menu.children('.dropdown-item').find('.action'),
+                    index = items.index(focused),
+                    prev = index - 1;
+
+                if (index === 0) {
+                    items.last().focus();
+                } else {
+                    items.eq(prev).focus();
+                }
+            },
+
+            nextMenuItemLink: function(focused, menu) {
+                var items = menu.children('.dropdown-item').find('.action'),
+                    items_count = items.length -1,
+                    index = items.index(focused),
+                    next = index + 1;
+
+                if (index === items_count) {
+                    items.first().focus();
+                } else {
+                    items.eq(next).focus();
+                }
+            },
+
+            focusFirstItem: function(menu) {
+                menu.find('.dropdown-item:first .action').focus();
+            },
+
+            closeDropdownMenu: function() {
+                var open = _dropdown.opts.page.find(_dropdown.opts.menu).not(':focus');
+
+                if (open) {
+                    open.removeClass(_dropdown.opts.menu_active)
+                        .addClass(_dropdown.opts.menu_inactive);
+
+                    open.parent()
+                        .find(_dropdown.opts.button)
+                        .removeClass(_dropdown.opts.button_active)
+                        .attr('aria-expanded', 'false');
+                }
+            },
+
+            closeDropdownMenus: function() {
+                var open = _dropdown.opts.page.find(_dropdown.opts.menu);
+
+                if (open) {
+                    open.removeClass(_dropdown.opts.menu_active)
+                        .addClass(_dropdown.opts.menu_inactive);
+
+                    open.parent()
+                        .find(_dropdown.opts.button)
+                        .removeClass(_dropdown.opts.button_active)
+                        .attr('aria-expanded', 'false');
+                }
+            },
+
+            openDropdownMenu: function(el) {
+                var menu = el.parent().find(_dropdown.opts.menu);
+
+                if (menu.hasClass(_dropdown.opts.menu_active)) {
+                    return false;
```

Comment:
Why return false? The return value is never used (nor documented). I would advise removing the if part of the condition.

In other works, just have an "if" condition with lines 158-165, if the menu does not have the class _dropdown.opts.menu_active.